pkg/virt-controller/watch/vm/vm.go

```diff
@@ -3191,16 +3191,42 @@ func (c *Controller) sync(vm *virtv1.VirtualMachine, vmi *virtv1.VirtualMachineI
 		return vm, common.NewSyncError(fmt.Errorf(fetchingRunStrategyErrFmt, err), FailedCreateReason), err
 	}
 
-	// Ensure we have ControllerRevisions of any instancetype or preferences referenced by the VM
-	err = c.instancetypeMethods.StoreControllerRevisions(vm)
-	if err != nil {
-		log.Log.Object(vm).Infof("Failed to store Instancetype ControllerRevisions for VirtualMachine: %s/%s", vm.Namespace, vm.Name)
-		c.recorder.Eventf(vm, k8score.EventTypeWarning, common.FailedCreateVirtualMachineReason, "Error encountered while storing Instancetype ControllerRevisions: %v", err)
-		return vm, common.NewSyncError(fmt.Errorf("Error encountered while storing Instancetype ControllerRevisions: %v", err), common.FailedCreateVirtualMachineReason), nil
+	// TODO(lyarwood): Extract all instance type logic into a handler
+	referencePolicy := c.clusterConfig.GetInstancetypeReferencePolicy()
+	switch referencePolicy {
+	case virtv1.Reference:
+		// Ensure we have ControllerRevisions of any instancetype or preferences referenced by the VM
+		if err = c.instancetypeMethods.StoreControllerRevisions(vm); err != nil {
+			log.Log.Object(vm).Infof("failed to store Instancetype ControllerRevisions for VirtualMachine: %s/%s", vm.Namespace, vm.Name)
+			c.recorder.Eventf(vm, k8score.EventTypeWarning, common.FailedCreateVirtualMachineReason, "Error encountered while storing Instancetype ControllerRevisions: %v", err)
+			return vm, common.NewSyncError(fmt.Errorf("Error encountered while storing Instancetype ControllerRevisions: %v", err), common.FailedCreateVirtualMachineReason), nil
+		}
+	case virtv1.Expand, virtv1.ExpandAll:
+		// Do not expand if there are no instance types and preferences or they already have a revisionName
+		if referencePolicy == virtv1.Expand && ((vm.Spec.Instancetype == nil && vm.Spec.Preference == nil) ||
+			(vm.Spec.Instancetype != nil && vm.Spec.Instancetype.RevisionName != "") ||
+			(vm.Spec.Preference != nil && vm.Spec.Preference.RevisionName != "")) {
```

Comment:
Should we at least log here (even on v4) to explain why we are not expanding?
I think it may be surprising for users choosing to switch to a new Expand policy to see that their VMs are not expanded.

Reply:
Yeah sure, I wasn't happy with the massive conditional here anyway so I'll extract and log something.

Reply:
https://github.com/kubevirt/kubevirt/pull/13086/commits/57643b2660dc0eb6b5a8417127d50595fca4c44b